designsafe/static/scripts/ng-designsafe/directives/ng-designsafe-directives.js

```diff
@@ -270,24 +270,26 @@ export function dsUserList(UserService) {
       usernames: '='
     },
     link: function (scope, element) {
-      var longString = '';
-      var promises = [];
+      scope.$watch('usernames', function() {
+        var longString = '';
+        var promises = [];
 
-      scope.usernames.forEach((usr) => {
-        promises.push(UserService.get(usr));
-      });
+        scope.usernames.forEach((usr) => {
```

Comment:
Would it make more sense to condense this into one call for all usernames, returning their corresponding first and last names as the response?

Reply:
> Would it make more sense to condense this into one call for all usernames, returning their corresponding first and last names as the response?

It would. `UserService.get` does not support multiple usernames at the moment, but I can put something together.